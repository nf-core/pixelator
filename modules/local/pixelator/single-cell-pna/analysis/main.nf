process PIXELATOR_PNA_ANALYSIS {
    tag "${meta.id}"
    label 'process_medium'

    // TODO: Add conda
    // conda "bioconda::pixelator=0.18.2"

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/pixelgen-technologies/pixelator:0.21.2'
        : 'quay.io/pixelgen-technologies/pixelator:0.21.2'}"

    input:
    tuple val(meta), path(data)

    output:
    tuple val(meta), path("analysis/*.pxl"),          emit: pixelfile
    tuple val(meta), path("analysis/*.report.json"),  emit: report_json
    tuple val(meta), path("analysis/*.meta.json"),    emit: metadata_json
    tuple val(meta), path("analysis/*"),              emit: all_results
    tuple val(meta), path("*pixelator-analysis.log"), emit: log

    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''

    """
    pixelator \\
        --cores ${task.cpus} \\
        --log-file ${prefix}.pixelator-analysis.log \\
        --verbose \\
        single-cell-pna \\
        analysis \\
        --output . \\
        ${args} \\
        ${data}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """


    stub:
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir "analysis"
    touch analysis/${prefix}.meta.json
    touch analysis/${prefix}.report.json
    touch analysis/${prefix}.pxl
    touch ${prefix}.pixelator-analysis.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
