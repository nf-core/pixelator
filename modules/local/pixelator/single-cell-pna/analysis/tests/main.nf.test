nextflow_process {

    name "Test Process PIXELATOR_PNA_ANALYSIS"
    script "../main.nf"
    process "PIXELATOR_PNA_ANALYSIS"
    tag "modules"
    tag "pixelator"
    tag "pixelator/pna"
    tag "pixelator/single_cell_pna_analysis"

    test("PNA analysis - small test") {
        when {
            process {
                """
                input[0] = [
                    [ id:'PNA055_Sample07_filtered_S7', design:'pna-2', panel:'proxiome-immuno-155', technology:'pna' ],
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/graph/PNA055_Sample07_filtered_S7.graph.pxl', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.metadata_json).match("metadata_json") },
                {
                    with (process.out.report_json) {
                        assert snapshot(path(get(0).get(1)).readLines()[0..5]).match("report_json") }
                },
                { assert snapshot(process.out.versions).match("versions") },
                {
                    with (process.out.log) {
                        assert path(get(0).get(1)).readLines().last().contains("Finished pixelator analysis")
                    }
                },
            )
        }
    }
}

