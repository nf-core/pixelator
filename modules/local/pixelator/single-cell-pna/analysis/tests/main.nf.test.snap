{
    "versions": {
        "content": [
            [
                "versions.yml:md5,cb111c2479c0716e77af331a52743451"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T11:58:37.130668357"
    },
    "report_json": {
        "content": [
            [
                "{",
                "    \"sample_id\": \"PNA055_Sample07_filtered_S7\",",
                "    \"product_id\": \"single-cell-pna\",",
                "    \"report_type\": \"analysis\",",
                "    \"proximity\": {},",
                "    \"k_cores\": {"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-22T17:31:47.907028898"
    },
    "metadata_json": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.meta.json:md5,2a6eaa66228775591525f74416f0b87f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-22T17:31:47.884573212"
    }
}